### PR TITLE
Adding PHP 8.2 & 8.3 (RC) images

### PR DESCRIPTION
**Pull request by codyfinegan**
Created new images for PHP 8.2 and 8.3. Both are based off of regular bullseye and not alpine (we can change them once the WIP alpine PR is good).

8.3 is based on the RC image and we'll update it once 8.3 is released, this is to get ahead on testing things out.

Python is removed from both as it's not needed.

### CODE REVIEW

File: compose/php.yml [hunk 1]
```diff
@@ -560,6 +560,106 @@ services:
...
+      - zsh-history:/root/.zsh_history
+      - ./shell:/root/custom_shell
+    depends_on:
+      - php-8.3-debug
+    networks:
+      - totara
+
+  php-8.3-cron:
+    image: totara/docker-dev-php83-cron
+    container_name: totara_php82_cron
```
Comment (derschatta) [inline compose/php.yml:634]: php83

File: php/php83-debug/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM totara/docker-dev-php83:latest
+
+RUN pecl install -f xdebug-3.3.0 && docker-php-ext-enable xdebug.so
```
Comment (derschatta) [inline php/php83-debug/Dockerfile:3]: I think 3.3.0 is not out fully, only `3.3.0alpha3`

File: README.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ Although this project started as a development environment for Totara Learn it c
 ### What You Get
  * [NGINX](https://nginx.org/) as a webserver
  * [Apache](https://httpd.apache.org/) as a webserver
- * [PHP](http://php.net/) 5.3, 5.4, 5.5, 5.6, 7.0, 7.1, 7.2, 7.3, 7.4, 8.0, 8.1 to test for different versions
+ * [PHP](http://php.net/) 5.3, 5.4, 5.5, 5.6, 7.0, 7.1, 7.2, 7.3, 7.4, 8.0, 8.1, 8.2, 8.3 to test for different versions
```
Comment (derschatta) [inline README.md:18]: 8.3RC

**Review by derschatta — CHANGES_REQUESTED** [review]
a few things to look at

**Comment by derschatta** [comment]
@codyfinegan  I've also created the repositories required on Docker Hub so theoretically they should build the images automatically once this gets merged

File: php/php83-debug/Dockerfile (hunk 2 above)
Comment (codyfinegan) [reply php/php83-debug/Dockerfile]: Yep, 3.3.0 installs the alpha version currently
> .789 WARNING: failed to download pecl.php.net/xdebug, version "3.3.0", will instead download version 3.3.0alpha3, stability "beta"

But it's better to lock into the alpha version & we update it after 8.3 goes live so I've changed to the alpha tag specifically

**Comment by codyfinegan** [comment]
> a few things to look at

Cheers, those are all updated.

**Review by derschatta — APPROVED** [review]

**Comment by derschatta** [comment]
looks good. Have not tested it yet. Want me to also build and test them?

**Comment by codyfinegan** [comment]
> looks good. Have not tested it yet. Want me to also build and test them?

Yes please. I've tested building locally (so all the images should build) but haven't had a chance to actually run them yet.

**Review by scq — APPROVED** [review]

**Comment by derschatta** [comment]
sorry couldn't get to testing this yet. @scq have you been able to?

**Comment by scq** [comment]
@derschatta Yup, I can confirm all 4 containers work

**Pull request merged**